tests/unit/tracing_test.cpp

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/test/tracing.h>
+}
+
+extern bool *traced_syscalls;
+extern char *traced_syscalls_str;
+
+class tracing_mock : public KernelSingleton {
+public:
+   MOCK_METHOD(int, handle_sys_trace_arg, (const char *arg), (override));
+};
+
+TEST(simple_wildcard_match, basic)
+{
+   const char *str1 = "test string";
+   const char *str1_equivalent = "test string*";
+   const char *qmark_str1 = "tes? string";
+   const char *wrong_str1 = "test string1";
+   const char *asterisk_str1 = "test *";
+   const char *str2 = "test strin1";
+   const char *invalid_str = "te*t ";
+
+   ASSERT_EQ(true, simple_wildcard_match(str1, str1));
+   ASSERT_EQ(true, simple_wildcard_match(str1, str1_equivalent));
+   ASSERT_EQ(true, simple_wildcard_match(str1, qmark_str1));
+   ASSERT_EQ(true, simple_wildcard_match(str1, asterisk_str1));
+   ASSERT_EQ(false, simple_wildcard_match(str1, str2));
+   ASSERT_EQ(false, simple_wildcard_match(str1, invalid_str));
+   ASSERT_EQ(false, simple_wildcard_match(str1, wrong_str1));
+}
+
+TEST(set_traced_syscalls_int, basic)
+{
+   tracing_mock mock;
+   const char *too_long = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```

Comment:
That's very bad.
1. Find where `TRACED_SYSCALLS_STR_LEN` and include that header here
2. Create an array on-stack like `const char too_long[TRACED_SYSCALLS_STR_LEN + 1];` and initialize it with memset.
3. Do the same for `almost_too_long`.